backend/datautil/timeutil/timeutils.py

```diff
@@ -142,7 +142,12 @@ def get_times(time_col):
             for hour_time in t_interval.split('-'):
                 # Add 'm' to convert it to Python recognizable format
                 hour_time += 'm'
-                formatted_interval = datetime.strptime(hour_time, '%I:%M%p')
+                try:
+                    formatted_interval = datetime.strptime(hour_time, '%I:%M%p')
+                except ValueError:
+                    print("[timeutils] Error interval when coverting to python time object: " + str(hour_time))
```

Comment:
Your error message has a typo in it in converting.

Reply:
So the reason I did %p was so that we could correctly determine whether the time was in AM or PM. Doing it like the way you did loses that context - be wary of making changes like this (I actually made some changes without understanding context when I first started at ServiceNow and thankfully my coworker caught me from causing major issues). In fact, if we get invalid input like this, it's probably better not to parse this at all.

In that case you should let the error propagate up and then in split into subsections just handle the error (give a good error message including all the info about the section), and then return an empty list of subsections. 

So I can do this or you can, I think it's a decent learning experience for you although I don't mind doing it myself. If you have any questions, just slack me on good error handling practices.

Reply:
Any suggests on what to print for each section, please?